### ISSUE TITLE
"incorrect header check" when sending a request with Axios

### ISSUE THREAD

**Issue by glpinho** [oven-sh/bun#4392]
### What version of Bun is running?

0.8.1

### What platform is your computer?

Linux 5.10.0-25-amd64 x86_64 unknown

### What steps can reproduce the bug?

Install Bun version 0.8.1 and send a HTTP request using Axios and using parameters (through AxiosRequestConfig<D = any>.params or in the URL).

### What is the expected behavior?

The request succeeds when using Bun version 0.6.1.

### What do you see instead?

error: incorrect header check
      at node:zlib:2358

### Additional information

Basically I started seeing these erros when I updated my Bun version to 0.8.1 (the latest available at the time I am writing this issue). It was working fine with Bun 0.6.1.

**Comment by Electroid** [comment]
Are you able to share the code that caused this?

**Comment by Jarred-Sumner** [comment]
Closing as stale. If you run into this, please let us know. Note that we are about to ship a rewrite to node:zlib.

**Comment by odinho** [comment]
Ran into this.

Biggest issue is that the _stack trace is so short_ it's not straight forward to tell what code caused it. This is a big old legacy app I'm trying to get running.

```

[fetch] > HTTP/1.1 GET https://storage.googleapis.com/storage/v1/b/mybucket.appspot.com/o/bucket%2Ffile.json?alt=media
[fetch] > accept-encoding: gzip
[fetch] > authorization: Bearer <REDACTED>
[fetch] > cache-control: no-store
[fetch] > user-agent: gcloud-node-storage/7.12.1
[fetch] > x-goog-api-client: gl-node/22.6.0 gccl/7.12.1-CJS gccl-invocation-id/6b1c1f71-71fb-484f-86c2-ef51d655fedb
[fetch] > Connection: keep-alive
[fetch] > Accept: */*
[fetch] > Host: storage.googleapis.com

[fetch] < 200 OK
[fetch] < ETag: CI3X+abRyYgDEAE=
[fetch] < X-Goog-Hash: crc32c=ZVPGGA==,md5=RHIR4yaDFmgRs/qz8PqC0A==
[fetch] < Content-Type: application/octet-stream
[fetch] < Content-Encoding: gzip
[fetch] < X-Goog-Stored-Content-Encoding: gzip
[fetch] < X-Goog-Stored-Content-Length: 2715235
[fetch] < X-Goog-Generation: 1726564318210957
[fetch] < X-Goog-Metageneration: 1
[fetch] < X-Goog-Storage-Class: STANDARD
[fetch] < Last-Modified: Tue, 17 Sep 2024 09:11:58 GMT
[fetch] < Pragma: no-cache
[fetch] < Expires: Mon, 01 Jan 1990 00:00:00 GMT
[fetch] < Cache-Control: no-cache, no-store, max-age=0, must-revalidate
[fetch] < Content-Disposition: attachment
[fetch] < Date: Sat, 28 Sep 2024 14:22:13 GMT
[fetch] < Vary: Origin
[fetch] < Vary: X-Origin
[fetch] < Vary: Accept-Encoding
[fetch] < X-GUploader-UploadID: AD-<ID>-A
[fetch] < Server: UploadServer
[fetch] < Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
[fetch] < Transfer-Encoding: chunked

1 | export default "native";
    ^
error: incorrect header check
      at onwrite (native:1:1)
      at native:1:1
      at _transform2 (native:1:1)
      at native:1:1
      at writeOrBuffer (native:1:1)
      at native:1:1
      at ondata (native:1:1)
      at emit (native:1:1)
      at addChunk (native:1:1)
      at readableAddChunk (native:1:1)

Bun v1.1.29 (Linux x64)
```

Tried with `--inspect-break` but it doesn't seem to catch this exception.

Something I know: Google Cloud Platform (storage) is *terrible* with regards to telling about encoding in its proprietary JSON API. When I implemented a emulator for faster tests, it was sometimes sending `gzip'ed` content but without telling the server about it. In a similar vein the code was expecting certainly encoded content back even if it wasn't explicitly stated.

So could be something simple like that. Just Google being really bad at writing code. Possibly. Or it could well be me :shrug:
